language/move-prover/boogie-backend-v2/src/prelude/vector-array-theory.bpl

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+// Basic theory for vectors using arrays. This version of vectors is not extensional.
+
+type {:datatype} Vec _;
+
+function {:constructor} Vec<T>(v: [int]T, l: int): Vec T;
+
+function {:builtin "MapConst"} MapConstVec<T>(T): [int]T;
```

Comment:
I just want to point out that `:builtin "MapConst"` is interpreted by Boogie only if /useArrayTheory is passed which indicates to Boogie that it should the native SMT array theory.

Reply:
Interesting, we have been using it before like this for ValueArray. I guess it doesn't matter for the non-extensional theories.